src/main/java/io/kestra/plugin/docker/Build.java

```diff
@@ -0,0 +1,218 @@
+import com.github.dockerjava.api.command.BuildImageCmd;
+import com.github.dockerjava.api.command.BuildImageResultCallback;
+import com.github.dockerjava.api.model.AuthConfig;
+import com.github.dockerjava.api.model.PushResponseItem;
+import com.github.dockerjava.core.DefaultDockerClientConfig;
+import io.kestra.core.models.annotations.PluginProperty;
+import io.kestra.core.models.tasks.RunnableTask;
+import io.kestra.core.models.tasks.Task;
+import io.kestra.plugin.scripts.exec.scripts.models.DockerOptions;
+import io.kestra.plugin.scripts.exec.scripts.runners.DockerService;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+import io.kestra.core.models.annotations.Example;
+import io.kestra.core.models.annotations.Plugin;
+import io.kestra.core.runners.RunContext;
+
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import javax.validation.constraints.NotNull;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+    title = "Execute a groovy script."
+)
+@Plugin(
+    examples = {
+        @Example(
+            title = "Build and push a docker image to a registry",
+            code = {"dockerfile: |",
+                "  FROM ubuntu",
+                "  ARG APT_PACKAGES=\"\"",
+                "",
+                "  RUN apt-get update && apt-get install -y --no-install-recommends ${APT_PACKAGES};",
+                "platforms:",
+                "- linux/amd64",
+                "tag: private-registry.io/unit-test:latest",
+                "buildArgs:",
+                "  APT_PACKAGES: curl",
+                "labels:",
+                "  unit-test: \"true\"",
+                "credential:",
+                "  username: <your-user>",
+                "  password: <your-password>",
+            }
+        ),
+    }
+)
+public class Build extends Task implements RunnableTask<Build.Output> {
+    @Schema(
+        title = "Docker api uri"
+    )
+    @PluginProperty(dynamic = true)
+    private String host;
+
+    @Schema(
+        title = "Credential for registry to push your image."
+    )
+    @PluginProperty
+    private DockerOptions.Credentials credentials;
+
+    @Schema(
+        title = "The templated string to render"
+    )
+    @PluginProperty(dynamic = true)
+    private String dockerfile;
+
+    @Schema(
+        title = "List of target platforms for build."
+    )
+    @PluginProperty(dynamic = true)
+    private List<String> platforms;
+
+    @Schema(
+        title = "Does the build will be push to a registry."
+    )
+    @PluginProperty(dynamic = true)
+    @Builder.Default
+    private Boolean push = false;
+
+    @Schema(
+        title = "Always attempt to pull a newer version of the base image."
+    )
+    @PluginProperty(dynamic = true)
+    @Builder.Default
+    private Boolean pull = true;
```

Comment:
just a note that I remember I proposed this initially and you mentioned it would be confusing to have both push and pull -- what made you change your mind to keep it?

Reply:
Maybe rename to `pullBaseImage` to be explicit

Reply:
Will kept `pull` to be coherent with GHA and docker cli